docs/proposals/pepxxx_wheel_variant_support.md

```diff
@@ -62,88 +62,650 @@ something they [indirectly do today](https://github.com/wheelnext/wheelnext/pull
+   of variant properties used by the wheel.
+
+3. As a JSON file called `{package}-{version}-variants.json` on the wheel index, where it includes a copy
+   of the aforementioned information that is used while determining which of the available wheel variants to install,
+   as well as list of all available wheel variants along with their respective properties. The `{package}`
+   and `{version}` templates correspond to the normalized project name and version, per the rules specified
+   in [binary distribution format
+   specification](https://packaging.python.org/en/latest/specifications/binary-distribution-format/).
+
+All these contexts use the same data structure that is defined in terms of Python dictionary below, and serialized
+using the standard TOML serialization, as sub-tables using the `[variant]` table as an anchor, or the standard JSON
+serialization as a top-level object.
+
+The following keys are defined:
+
+- `providers` containing a dictionary of used variant providers, with the names of their namespaces as keys
+
+- `default-priorities` specifying the default priorities used to order wheels by preference
+
+- `variants` containing a dictionary of wheel variants, with their labels as keys and properties as values
+  (not present in `pyproject.toml`)
+
+#### Provider information
+
+The provider information dictionary provides information on how to install and use variant providers. It must be
+specified in `pyproject.toml` for every variant namespace that is supported. It must be copied to `variant.json` as-is,
+including data for providers that are not used in the particular wheel. The dictionary keys are namespaces, while
+the values are dictionaries.  They must include the following key:
+
+- `requires: list[str]` specifying a list of one or more package dependency specifiers. When installing the provider,
+  all the dependencies are installed (provided their environment markers match).
+
+Additionally, they may include the following keys:
+
+- `enable-if: str` specifying an environment marker defining when the plugin should be used. If the environment marker
+  does not match the running environment, the provider will be disabled and the variants using its properties will
+  deemed incompatible.
+
+- `optional: bool` specifying whether the provider is optional, as a boolean value. If it is true, the provider
+  is considered optional and it should not be used unless the user opts in to it, effectively rendering the variants
+  using its properties incompatible. If it is false or missing, the provider is considered required.
+
+- `plugin-api: str` specifying the API endpoint for the plugin. If it is specified, it must be an object reference
+  as explained in the "API endpoint" section. If it is missing, the package name from the first dependency specifier
+  in `requires` is used, after replacing all the `-` characters with `_`.
```

Comment:
```suggestion
  in `requires` is used, after replacing all `-` characters with `_` in the normalized package name.
```

Spec ref: https://packaging.python.org/en/latest/specifications/binary-distribution-format/#escaping-and-unicode, this doesn't seem to have a name in the spec. The text change normalizes for potential `.` too.